File: spr_2/Ćwiczenie 1 – Gnatowski_ulamek.cpp
/*
 * Ćwiczenie 1 – Gnatowski_ulamek.cpp
 * 
 */

#include <iostream>

using namespace std;

int nwd(int a, int b)
{
    while(a > 0)
    {
        a = a%b;
        b -= a;
    }
    return b;
}


int main(int argc, char **argv)
{
    int li = 0;
    int mia = 0;
    
    cout<<"Podaj licznik: ";
    cin>>li;
    cout<<"Podaj mianowkik: ";
    cin>>mia;
    
    cout<<"Ulamek po skroceniu: "<<(li/nwd(li,mia))<<'/'<<(mia/nwd(li,mia));
    return 0;
}

File: spr/Ćwiczenie 1 – Gnatowski_ulamek.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def nwd(a, b):
    while a > 0:
        a = a % b
        b = b - a
    return b


def main(args):
    li = int(input("Podaj licznik: "))
    mia = int(input("Podaj mianownik: "))

    assert nwd(1989, 867) == 51

    print("Ulamek po skroceniu: ",(li/nwd(li,mia)), '/', (mia/nwd(li,mia)))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
